kubernetes-client/src/com/goyeau/kubernetes/client/api/PodsApi.scala

```diff
@@ -53,66 +56,67 @@ private[client] case class NamespacedPodsApi[F[_]](httpClient: Client[F], config
     with Watchable[F, Pod] {
   val resourceUri: Uri = uri"/api" / "v1" / "namespaces" / namespace / "pods"
 
-  def exec[Result](
+  val execHeaders: Headers =
+    Headers.of(
+      config.authorization.toList ++ headers.`Sec-WebSocket-Protocol`.parse("v4.channel.k8s.io").toOption.toList: _*
+    )
+
+  val webSocketAddress: Uri = Uri.unsafeFromString(
+    config.server
+      .toString()
+      .replaceFirst("http", "ws")
+  )
+
+  def exec[T: Monoid](
       podName: String,
-      flow: Flow[Either[Status, String], Message, Future[Result]],
+      flow: ExecStream => T,
```

Comment:
I am not sure how user would like to consume server messages. This flow function should allow user to capture message via Monoid, potentially discarding some them or converting into string. 🤔